variable/test/comparison_test.cpp

```diff
@@ -70,3 +71,37 @@ TEST(ComparisonTest, tolerance_type_mismatch) {
   EXPECT_THROW(is_approx(a, b, 0 /*implicit int*/), except::TypeError);
   EXPECT_NO_THROW(is_approx(a, b, 0.0f));
 }
+
+TEST(ComparisonTest, less_variances_test) {
+  const auto a = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0},
+                                     Variances{1.0, 1.0});
+  const auto b = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{0.0, 3.0},
+                                     Variances{1.0, 1.0});
+  EXPECT_THROW(less(a, b), std::runtime_error);
+}
+
+TEST(ComparisonTest, less_dtypes_test) {
+  const auto a = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0});
+  const auto b = makeVariable<int>(Dims{Dim::X}, Shape{2}, Values{0, 1});
+  EXPECT_THROW(less(a, b), std::runtime_error);
+}
+
+TEST(ComparisonTest, less_units_test) {
+  const auto a = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0});
+  auto b = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{0.0, 3.0});
+  b.setUnit(units::m);
+  EXPECT_THROW(less(a, b), std::runtime_error);
+}
+
+const auto a = 1.0 * units::Unit(units::m);
+const auto b = 2.0 * units::Unit(units::m);
+const auto true_ = true * units::Unit(units::dimensionless);
+const auto false_ = false * units::Unit(units::dimensionless);
```

Comment:
Put in anonymous namespace to avoid linker errors if, e.g., `a` defined in a different translation unit? Alternatively use a test fixture and define these in there.